simulator/infra/instrcache/instr_cache_memory.h

```diff
@@ -55,11 +55,17 @@ class InstrMemory : public FuncMemory
         }
 
         template<typename T>
-        void write(T value, Addr addr, T mask = all_ones<T>())
+        void mem_write( T value, Addr addr, T mask = all_ones<T>())
```

Comment:
I don't like the new naming. We are working with memory class already, so that's obvious we work with memory.